### PR TITLE
Non flash MPT.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

This adds a non flash version of MPT.
Flash is harder because we need to create a bias ready cuda kernel of flash attention.

Fixes https://github.com/huggingface/text-generation-inference/issues/361
Fixes https://github.com/huggingface/text-generation-inference/issues/491
Fixes https://github.com/huggingface/text-generation-inference/issues/290

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: server/text_generation_server/models/custom_modeling/mpt_modeling.py [hunk 1]
```diff
@@ -0,0 +1,1173 @@
...
+                pos = torch.clamp(
+                    pos
+                    - torch.cumsum((~attention_mask).to(torch.int32), dim=1)[
+                        :, past_position:
+                    ],
+                    min=0,
+                )
+            pos_emb = self.wpe(pos)
+            x = tok_emb + pos_emb
+        # if self.embedding_fraction == 1:
```
Comment (OlivierDehaene) [inline server/text_generation_server/models/custom_modeling/mpt_modeling.py:1005]: Remove?

File: server/text_generation_server/models/mpt.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+        filename = hf_hub_download(model_id, revision=revision, filename="config.json")
+        with open(filename, "r") as f:
+            config = json.load(f)
+        config = PretrainedConfig(**config)
+        config.quantize = quantize
+        # config = AutoConfig.from_pretrained(
+        #     # model_id, revision=revision, trust_remote_code=trust_remote_code
+        #     model_id, revision=revision, trust_remote_code=False
+        # )
```
Comment (OlivierDehaene) [inline server/text_generation_server/models/mpt.py:71]:
```suggestion
```

**Review by OlivierDehaene — COMMENTED** [review]
There are some comments in the mpt_modeling file. Should they be removed?

File: server/text_generation_server/models/mpt.py (hunk 2 above)
Comment (Narsil) [reply server/text_generation_server/models/mpt.py]: Done.

File: server/text_generation_server/models/custom_modeling/mpt_modeling.py (hunk 1 above)
Comment (Narsil) [reply server/text_generation_server/models/custom_modeling/mpt_modeling.py]: Done.

**Pull request merged**